hive-metastore/src/main/java/org/apache/iceberg/hive/HiveSchemaUtil.java

```diff
@@ -75,7 +91,46 @@ public static Schema convert(List<FieldSchema> fieldSchemas, boolean autoConvert
       typeInfos.add(TypeInfoUtils.getTypeInfoFromTypeString(col.getType()));
       comments.add(col.getComment());
     }
-    return HiveSchemaConverter.convert(names, typeInfos, comments, autoConvert);
+    Schema schema = HiveSchemaConverter.convert(names, typeInfos, comments, autoConvert);
+    return rebuildSchemaWithIdentifierFieldIds(schema, identifierFieldNames);
+  }
+
+  /**
+   * Rebuild a schema with given schema and identifierFieldNames
+   * @param schema The origin schema.
+   * @param identifierFieldNames The identifierFieldNames.
+   * @return New schema with IdentifierFieldIds.
+   */
+  @VisibleForTesting
+  static Schema rebuildSchemaWithIdentifierFieldIds(Schema schema, Set<String> identifierFieldNames) {
+    if (identifierFieldNames.size() == 0) {
+      return schema;
+    }
+    // Identifier fields in nested field are not supported, so we just check the first level columns.
```

Comment:
nit: extra new line

Reply:
Can't we give a better error message by checking if the request has a nested field and rejecting it, rather than only checking for top-level columns?